src/creatures/players/player.cpp

```diff
@@ -2469,6 +2469,91 @@ void Player::death(Creature* lastHitCreature)
 		onIdleStatus();
 		sendStats();
 	}
+	despawn();
+}
+
+bool Player::spawn()
+{
+	setDead(false);
+
+	const Position& pos = getLoginPosition();
+
+	if (!g_game.map.placeCreature(pos, this, false, true)) {
+		return false;
+	}
+
+	SpectatorHashSet spectators;
+	g_game.map.getSpectators(spectators, position, false, true);
+	for (Creature* spectator : spectators) {
```

Comment:
It's interesting to add a checker here, 
```suggestion
	for (Creature* spectator : spectators) {
	    if (!spectator) {
	        continue;
	    }
```